crates/llm/src/lib.rs

```diff
@@ -103,59 +107,67 @@ pub enum ModelArchitecture {
     /// [GPT-NeoX](llm_neox)
     NeoX,
 }
-impl Display for ModelArchitecture {
+
+/// An unsupported model architecture was specified
+pub struct UnsupportedModelArchitecture(String);
```

Comment:
I suggest implementing `std::error::Error` for this. Shouldn't need to implement any of its methods.